bolt-sidecar/src/driver.rs

```diff
@@ -224,24 +248,41 @@ impl<C: StateFetcher, BLS: SignerBLS, ECDSA: SignerECDSA> SidecarDriver<C, BLS,
         // parse the request into constraints and sign them
         let slot = inclusion_request.slot;
 
+        let pubkey = match self.constraint_signer {
```

Comment:
reminder: we're missing a check here. this pubkey is used to sign constraints but we don't know if it is authorized to do so. this can be added after we load delegation info in #290